instrumentation/jaxrs/jaxrs-1.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JerseyTest.java

```diff
@@ -0,0 +1,127 @@
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class JerseyTest extends AbstractHttpServerUsingTest<Server> {
+
+  @RegisterExtension
+  public static final InstrumentationExtension testing =
+      HttpServerInstrumentationExtension.forAgent();
+
+  @BeforeAll
+  protected void setUp() {
+    startServer();
+  }
+
+  @AfterAll
+  protected void cleanUp() {
+    cleanupServer();
+  }
+
+  @Override
+  protected Server setupServer() throws Exception {
+    ServletContainer servlet =
+        new ServletContainer(
+            new Application() {
+              @Override
+              public Set<Object> getSingletons() {
+                Set<Object> objects = new HashSet<>();
+                objects.add(new Resource.Test1());
+                objects.add(new Resource.Test2());
+                objects.add(new Resource.Test3());
+                return objects;
+              }
+            });
+
+    ServletContextHandler handler = new ServletContextHandler(ServletContextHandler.SESSIONS);
+    handler.setContextPath(getContextPath());
+    handler.addServlet(new ServletHolder(servlet), "/*");
+
+    Server server = new Server(port);
+    server.setHandler(handler);
+    server.start();
+
+    return server;
+  }
+
+  @Override
+  protected void stopServer(Server server) throws Exception {
+    server.stop();
+    server.destroy();
+  }
+
+  @Override
+  protected String getContextPath() {
+    return "/jetty-context";
+  }
+
+  private static Stream<Arguments> provideArguments() {
+    return Stream.of(
+        Arguments.of("test/hello/bob", "/test/hello/{name}", "Test1", "hello", "Test1 bob!"),
+        Arguments.of("test2/hello/bob", "/test2/hello/{name}", "Test2", "hello", "Test2 bob!"),
+        Arguments.of("test3/hi/bob", "/test3/hi/{name}", "Test3", "hello", "Test3 bob!"),
+        Arguments.of("test3/nested", "/test3/nested", "Test3", "nested", "Test3 nested!"));
+  }
+
+  @ParameterizedTest
+  @MethodSource("provideArguments")
+  void request(
+      String resource,
+      String expectedRoute,
+      String className,
+      String methodName,
+      String expectedResponse) {
+    AggregatedHttpResponse response =
+        client.post(address.resolve(resource).toString(), "bob").aggregate().join();
+    assertThat(response.status().code()).isEqualTo(200);
+    assertThat(response.contentUtf8()).isEqualTo(expectedResponse);
+
+    String controllerName = className + "." + methodName;
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("POST " + getContextPath() + expectedRoute)
+                        .hasKind(SpanKind.SERVER)
+                        .hasNoParent()
+                        .hasAttributesSatisfying(
```

Comment:
are we intentionally avoiding using `hasAttributesSatisfyingExactly` here so that we don't need to to assert all the other attributes on this span? Also the groovy tests [expected](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/71838a63d243d950156c689019e6621c227d45e9/instrumentation/jaxrs/jaxrs-1.0/javaagent/src/test/groovy/JerseyTest.groovy#L88) an `error.type` that I don't see on this test's spans, is that an artifact of us switching the test dependencies from using the dropwizard testing package to using jetty directly?

Reply:
Yes I intentionally skipped listing all attributes, but I guess they could be added for consistency with other tests. I was also confused by the `error.type` it is added because `runWithHttpServerSpan` uses https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/e27a76afca88f74b5156c54db91911734ac27165/testing-common/src/main/java/io/opentelemetry/instrumentation/testing/TestInstrumenters.java#L138 that returns `null` from `getHttpResponseStatusCode` and https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/e27a76afca88f74b5156c54db91911734ac27165/instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/semconv/http/HttpCommonAttributesExtractor.java#L106 interprets `null` as non successful request. I suspect this behavior might not conform to the spec as the spec says `If the request has completed successfully, instrumentations SHOULD NOT set error.type`. Does lack of status code mean that the request has failed? Imo not too important as real instrumentations always set the status code, and if they don't then there really must have been an error. cc @trask

Reply:
I think it's ok to interpret `null` response status code as "not completed successfully", since a "successful" HTTP response would always have a response status code